pyproject.toml

```diff
@@ -79,20 +79,44 @@ ignore = [
   "E722"
 ]
 
-[tool.ruff.lint]
-preview = true
-select = [
-# Defaults
-"E4", "E7", "E9", "F",
-# Undefined export
-"F822",
-# Useless import alias
-"PLC0414"
-]
 
-ignore = [
-  # Module import not at top of file
-  "E402",
-  # Do not use bare `except`
-  "E722"
+[tool.mypy]
+files = ["array_api_compat"]
+python_version = "3.10"
+disallow_incomplete_defs = true
+disallow_untyped_decorators = true
+disallow_untyped_defs = false  # TODO
+ignore_missing_imports = true
+no_implicit_optional = true
+show_error_codes = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+warn_unreachable = true
+
+[[tool.mypy.overrides]]
+module = ["cupy.*", "dask.*", "jax.*", "ndonnx.*", "sparse.*", "torch.*"]
+ignore_missing_imports = true
+
+
+[tool.pyright]
+include = ["src", "tests"]
+pythonVersion = "3.10"
```

Comment:
this isn't needed

```suggestion
```

it also makes type-checking in a CI matrix of different python version pretty much useless

Reply:
removed